azure/src/test/java/com/microsoft/azure/management/TestContainerService.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management;
+
+import com.microsoft.azure.management.compute.*;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import org.apache.commons.codec.binary.Base64;
+import org.junit.Assert;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.interfaces.RSAPublicKey;
+
+public class TestContainerService extends TestTemplate<ContainerService, ContainerServices> {
+
+    @Override
+    public ContainerService createResource(ContainerServices containerServices) throws Exception {
+        final String sshKeyData =  this.getSSHKey();
+
+        final String newName = "as" + this.testId;
+        final String dnsPrefix = "dns" + newName;
+        ContainerService containerService = containerServices.define(newName)
+                .withRegion(Region.US_EAST)
+                .withNewResourceGroup()
+                .withMasterProfile(ContainerServiceMasterProfileCount.MIN, "mp1" + dnsPrefix)
+                .withLinuxProfile()
+                .withRootUsername("testUserName")
+                .withSshKey(sshKeyData)
+                .defineContainerServiceAgentPoolProfile("agentPool0" + newName)
+                    .withCount(1)
+                    .withVmSize(ContainerServiceVMSizeTypes.STANDARD_A1)
+                    .withDnsLabel("ap0" + dnsPrefix)
+                .attach()
+                .withDCOSOrchestration()
+                .withDiagnostics()
+                .withTag("tag1", "value1")
+                .create();
+        return containerService;
```

Comment:
need Asserts verifying the creation went as intended, testing each specified property.